fix(WAKU2-RPC): add clarification for base64 encoding format

LNSD opened the pull request:
This PR amends the RFC:
- [x] Clarifying the **Base64** encoding format.
- [x] Provide a link to the IETF RFC 4648's relevant section.

Commit:
fix(WAKU2-RPC): add clarification for base64 encoding format

richard-ramos commented:
I'm curious on why was URL Safe Base64 encoding chosen instead of the standard? 
Reason why I ask is because i noticed that the base64 encoding used in go is the Std encoding https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/encoding/json/encode.go;l=839

richard-ramos commented:
I'm wondering if we could change the RFC + Implementation to use the standard encoding instead,  because if i understand correctly in JS https://developer.mozilla.org/en-US/docs/Glossary/Base64 btoa and atob use standard base64 encoding, meaning that to use base64url encoding it would require an additional step of doing replacing + => - and / => _ and removing extra = from the end of the encoded string, or use an npm package like base64url to do that.

**Edit:**
This is by no means a blocker for me. I can create a custom marshaling function in go-waku to use that particular base64 encoding. I'm just curious to know if there was a particular reason for choosing this format over std base64

richard-ramos commented:
Copied from Discord:
> @fryorcraken: I am not sure btoa is relevant in this context as I don't expect hitting the JSON RPC API from a browser
> @rramos: how do you consume rpc api in js-waku? is it via nodejs?
> @fryorcraken: yes, purely in NodeJS to start the nodes and set it up before running the tests.

richard-ramos commented:
@fryorcraken, do you know what encoding is used when marshalling a byte array field in a protobuffer to JSON in JS?
I see that buffers support both 'base64' and 'base64url', but I'm curious to know what is the format chosen by the protobuffer package used by js-waku

fryorcraken commented:
> marshalling a byte array field in a protobuffer to JSON in JS?

Not sure I get it. When decoding a protobuf message in JSON, we get a `Uint8Array` which is the native browser type for byte arrays.

richard-ramos commented:
I didn't explain myself correctly, my apologies. I'll try to do a practical example instead:

In go, I have the following struct which represents a WakuMessage protobuffer:
```go
type WakuMessage struct {
        // In this example i'm only defining the payload
	Payload        []byte          `protobuf:"bytes,1,opt,name=payload,proto3" json:"payload,omitempty"`
}
```
I can obtain a json string from that WakuMessage by doing the following:
```go
myWakuMessage := &WakuMessage{ 
    Payload: []byte("Hello World"),
}
jsonBytes, err := json.Marshal(myWakuMessage)
fmt.Println(string(jsonBytes))

// The result is `{"payload":"SGVsbG8gV29ybGQ="}`
```
The `payload` attribute from the json contains the result of serializing the WakuMessage's `Payload` byte array into a string using standard `base64`. This same serialization technique would be applied if I were to execute a call from the JSON RPC API unless I define a custom marshaller that uses `base64url` encoding

I was curious on how would this work in JSLand with the protobuffer library used in js-waku, but maybe this is not something that's usually done there!

fryorcraken commented:
This functionality does not exists in `protons` (the lib we use with JS). Using `toString` would lead the byte field to be printer as an array of numbers `[ 1, 2, 3 ]`

LNSD commented:
Ok... There was some confusion on my side: I thought the payload was encoded in base64 URL-friendly format without padding. This is not what was happening. It is encoded in **regular base64 with padding**. It seems I implemented it the right way from the very beginning.

So the v0.15.0 is working as expected: **Payload must be encoded using regular base64 (with padding)**.

I have updated this PR so it is clear that the encoding must be **with padding**.
